model/host/host.go

```diff
@@ -341,6 +342,27 @@ func (h *Host) SetDNSName(dnsName string) error {
 	return err
 }
 
+func (h *Host) SetIPv6Address(ipv6Address string) error {
+	err := UpdateOne(
+		bson.M{
+			IdKey:  h.Id,
+			IPv6Key: "",
+		},
+		bson.M{
+			"$set": bson.M{
+				IPv6Key: ipv6Address,
+			},
+		},
+	)
+	if err == nil {
+		h.IPv6 = ipv6Address
+	}
```

Comment:
it's typically a better plan to handle the error situations first and then do the normal cases below. Certianly easier to modify later. I'd refactor this function.